### PR TITLE
Ra 89 - fixes

### PR DESCRIPTION
**Pull request by aameijide**
* Add Fody for Oberservable Collection
* Delete App.Registry
* Fix test

### CODE REVIEW

File: Weavers/Forwarder.cs [hunk 1]
```diff
@@ -110,10 +121,24 @@ bool IsBindableBaseType (TypeDefinition type)
...
+			}
+		}
+
 		void ProcessType (TypeDefinition type)
 		{
 			LogInfo ("\t" + type.FullName);
 			foreach (var property in type.Properties) {
+
 				if (!bindableTypes.Contains (property.PropertyType.Resolve ()) &&
 				    !property.PropertyType.Name.StartsWith ("ObservableCollection")) {
```
Comment (ylatuya) [inline Weavers/Forwarder.cs]: This should check in observableTypes.

File: VAS.Core/Store/MediaFileSet.cs [hunk 2]
```diff
@@ -28,7 +29,7 @@ namespace VAS.Core.Store
 	[Serializable]
 	[PropertyChanged.ImplementPropertyChanged]
 	[JsonObject]
-	public class MediaFileSet : List<MediaFile>, IStorable
+	public class MediaFileSet : ObservableCollection<MediaFile>, IStorable
```
Comment (ylatuya) [inline VAS.Core/Store/MediaFileSet.cs]: Add aunit test to the check projects emit the event when mediafileset is changed

File: Weavers/Forwarder.cs (hunk 1 above)
Comment (aameijide) [inline Weavers/Forwarder.cs]: observableTypes are added to bindableTypes in linea 103, so bindableTypes now has BindableBase and ObservableCollection types, it means, it checks both.

File: Weavers/Forwarder.cs [hunk 3]
```diff
@@ -91,7 +96,13 @@ public void Execute ()
 			bindableTypes = new HashSet<TypeDefinition> (
 				ModuleDefinition.GetTypes ().Where (x => x.IsClass && IsBindableBaseType (x)));
 
-			// Process BindableBase types to inject the ConnectChild () function
+			// Get a list of all the classes deriving from ObservableCollection
+			var observableTypes = new HashSet<TypeDefinition> (
+				                      ModuleDefinition.GetTypes ().Where (x => x.IsClass && IsObservableCollectionType (x)));
+
+			observableTypes.ToList ().ForEach (x => bindableTypes.Add (x));
```
Comment (ylatuya) [inline Weavers/Forwarder.cs]: ToList() shouldn't be needed here

Comment (aameijide) [inline Weavers/Forwarder.cs]: ToList() was added in order to use linq. I change it for the normal foreach:
			foreach (var x in observableTypes) {
				bindableTypes.Add (x);
			}

File: Weavers/Forwarder.cs (hunk 1 above)
Comment (ylatuya) [inline Weavers/Forwarder.cs]: So this check is no longer needed  !property.PropertyType.Name.StartsWith ("ObservableCollection")

Comment (aameijide) [inline Weavers/Forwarder.cs]: We shouldn't. If we delete this ObservableCollection (not inherited classes) are not setted

**Comment by aameijide** [comment]
Ready to review again

File: VAS.Core/Common/StateController.cs [hunk 4]
```diff
@@ -213,6 +213,9 @@ public void Register (string transition, Func<IScreenState> panel)
 		/// <param name="transition">Transition.</param>
 		public bool UnRegister (string transition)
 		{
+			// Delete all existing instances
+			navigationStateStack.Where (x => x.Item1 == transition).ToList ().ForEach (t => t.Item2.Panel.Dispose ());
```
Comment (ylatuya) [inline VAS.Core/Common/StateController.cs]: There can be only one transition with this name, use First()

File: VAS.Core/Common/StateController.cs [hunk 5]
```diff
@@ -243,6 +246,7 @@ public bool UnRegister (string transition)
```
Comment (ylatuya) [inline VAS.Core/Common/StateController.cs]: EmptyStateStack is clearing the stack without disposing elements.

Update unit test to check that dispose is called correctly.

File: Weavers/Forwarder.cs (hunk 1 above)
Comment (ylatuya) [inline Weavers/Forwarder.cs]: That is still not fixed.

Comment (ylatuya) [inline Weavers/Forwarder.cs]: That's still not fixed

File: VAS.Core/Common/StateController.cs (hunk 5 above)
Comment (ylatuya) [inline VAS.Core/Common/StateController.cs]: That's still not fixed

File: Weavers/Forwarder.cs (hunk 1 above)
Comment (aameijide) [inline Weavers/Forwarder.cs]: I think is not possible, to delete property.PropertyType.Name.StartsWith ("ObservableCollection"). When a property is of type ObservableCollection<Something>, we have to compare with the same time (ObservableCollection<Something>) and is not possible no compare it with ObservableCollection neither ObservableCollection<T>, so I think that currently we have the best approach.

**Comment by aameijide** [comment]
Ready to review again

**Pull request merged**